File: ConcMath1.java
import java.math.*;
import java.util.*;
import java.util.concurrent.*;

public class ConcMath1 {
    private static List<Integer> calcPrimeFactors(int n) {
        List<Integer> pfs = new ArrayList<Integer>();

	int d = 2;
	while (d*d < n) {
	    while (n%d == 0) {
	        pfs.add(d);
		n = n/d;
	    }
	    d++;
	}
	if (n > 1) {
	    pfs.add(n);
	}

	return pfs;
    }

    public static void main(String [] args) throws Exception {
        long start = new Date().getTime();
        List<Thread> tlist = new ArrayList<Thread>();
	Random rand = new Random();
        for (int batch = 0; batch < 100; batch++) {
	    Thread t = new Thread(() -> {
	        for (int i = 0; i < 10000; i++) {
		    int rnum = rand.nextInt(9800000) + 20000;
		    List<Integer> pfs = calcPrimeFactors(rnum);
		    //System.out.println(rnum + "->" + pfs);
		}
	    });
	    tlist.add(t);
	    t.start();
	}
	for (Thread t : tlist) {
	    t.join();
	}
        long end = new Date().getTime();
	System.out.println("Time taken is " + (end - start) + " ms");
    }
}

File: concmath1.py
#!/usr/bin/env python3

import time
import random
from threading import Thread

def calcPrimeFactors(n):
    primeFactors = []
    d = 2
    while d*d < n :
        while (n%d == 0) :
            primeFactors.append(d)
            n //= d
        d = d+1
    if n > 1 :
       primeFactors.append(n)
    return primeFactors

def batch():
    for i in range(10000):
        rand = random.randint(20000, 10000000)
        pfs = calcPrimeFactors(rand)
        #print (rand, pfs)

def main() :
    start = time.time()
    threads = []
    for i in range(100) :
        thread = Thread(target=batch)
        threads.append(thread)
        thread.start()
    for t in threads:
        t.join()
    end = time.time()
    print (f"Execution time is {end-start} sec")
    
if __name__ == "__main__" :
    main()
